### PR TITLE
Jenkins 54352

### PR DESCRIPTION
**Pull request by kevinearls**
This PR adds two test cases which build the JFR using JCasC.  One adds a maven instance, the other adds a global pipeline library.

This should not be merged until we resolve a few issues.

1. Is there a better way to specify casc.yml files in the packager-config.yml files, and not have to copy them?  
2. Can the casc.yml code for test_cwp_casc_with_tool be updated to install maven?
3. I'm searching for a  b bbetter global library to use for test_cwp_casc_with_pipeline_library, as the current tests will behave differently depending on where it is run.

### CODE REVIEW

**Comment by kevinearls** [comment]
@varyvol @fcojfernandez Please review.

File: tests/test_resources/test_cwp_casc_with_pipeline_library/packager-config.yml [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+    artifactId: "configuration-as-code-support"
+    source:
+      version: "1.5"
+systemProperties: {
+    jenkins.model.Jenkins.slaveAgentPort: "50000",
+    jenkins.model.Jenkins.slaveAgentPortEnforce: "true"}
+casc:
+  - id: "casc"
+    source:
+      dir: casc.yml
```
Comment (varyvol) [inline tests/test_resources/test_cwp_casc_with_pipeline_library/packager-config.yml:105]: If you specify "./casc.yml" maybe you wouldn't need to copy the file first?

**Review by varyvol — COMMENTED** [review]
Some comments:
- Regarding 1), I've made a suggestion. Maybe it works like that.
- I'm not sure but if you pick a "default version" maybe the tool configuration is able to auto install the tool.
- I don't think pipeline-infra is a bad global library at all. If I'm not wrong, even when the test is executed on CI, since this a a JFR image, the result should not be different.

By the way, I think these tests should reside in the CWP repository.

**Review by fcojfernandez — COMMENTED** [review]
I think these tests might be a bit redundant here and maybe it makes more sense to move them to the CWP repository. Instead of downloading the jar or using the docker image, you might generate a local jar file and test the JFR image generated, so you would be testing the master branch of CWP.

**Pull request closed without merging**